### PR TITLE
Add a stricter test for stoppableFoldl behaviour

### PR DESCRIPTION
**Pull request by Janiczek**
@adamnfish has found that the tests for `List.Extra.stoppableFoldl` aren't really testing the main promise of the function: that it will stop running the function after a `Stop` is returned.

Together we're adding a test which will throw a RangeError (stack overflow) if `stoppableFoldl` goes through the whole list. We've verified it by temporarily changing the implementation of `stoppableFoldl` to just be a `List.foldl`.

### CODE REVIEW

**Comment by Chadtech** [comment]
Nice catch! Thank you.

File: tests/Tests.elm [hunk 1]
```diff
@@ -993,5 +993,30 @@ all =
...
+                    stoppableFoldl
+                        (\n acc ->
+                            if n < 50 then
+                                Continue n
+
+                            else if n == 50 then
+                                Stop n
+
+                            else
+                                explode ()
```
Comment (Janiczek) [inline tests/Tests.elm:1016]: @Chadtech I've just realized this probably is missing a function name change 😰  `explode` can't be found anywhere, if my memory's correct it was supposed to be `throwRangeErrorException ()` instead of `explode ()`. This will likely not compile.

Sorry, I guess a commit got lost / didn't get pushed / ... I'll create a fix PR

Comment (Janiczek) [reply tests/Tests.elm:1016]: Actually the [last version of the code](https://github.com/elm-community/list-extra/blob/master/tests/Tests.elm#L1131) has `throwRangeErrorException ()` already. Is this some kind of GitHub glitch? Weird. (EDIT: ah, it was fixed in #163) I guess we don't need to do anything then

Comment (Chadtech) [reply tests/Tests.elm:1016]: No worries! I noticed the code wasn't compiling after I merged it in, but it was an easy fix.

Thanks for your original contribution, and thanks for noticing now!
